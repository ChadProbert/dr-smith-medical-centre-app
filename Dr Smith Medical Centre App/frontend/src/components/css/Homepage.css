/* imported google fonts */
@import url("https://fonts.googleapis.com/css?family=Padauk");
@import url("https://fonts.googleapis.com/css?family=Open+Sans");
@import url("https://fonts.googleapis.com/css?family=Roboto+Slab");

.welcomeHeading,
.welcomeSubHeading {
	/* applied the google font */
	font-family: "Padauk", sans-serif;
	margin-left: 100px;
	text-align: left;
}

.welcomeHeading {
	font-size: 75px;
	font-weight: 700;
	margin-top: 100px;
}

.aboutUsBox {
	background-color: rgba(236, 236, 236, 0.658);
	padding-left: 25%;
	padding-right: 25%;
	text-align: center;
	padding-top: 120px;
	padding-bottom: 120px;
	/* applied Google Font */
	font-family: "Open Sans", sans-serif;
}

.aboutUsHeading,
.reviewsHeading {
	font-weight: bold;
	margin-bottom: 30px;
}

/* used !important to overwrite default config set by MaterialUI */
.appointmentsButton {
	margin-top: 20px !important;
	margin-left: 100px !important;
	margin-bottom: 30px !important;
	background-color: rgb(53, 192, 237) !important;
}

.adminButton {
	margin-left: 100px !important;
	background-color: rgb(40, 194, 127) !important;
	margin-bottom: 120px !important;
}

/* removes the blue underline applied to links */
.appointmentsLink,
.adminLoginLink {
	text-decoration: none;
}

/* applied a background image to the welcome and about us section */
.background1 {
	background-image: url("../images/background images/11.jpg");
	background-size: cover;
	background-repeat: no-repeat;
}

.reviewsBox {
	padding-left: 10%;
	padding-right: 10%;
	/* vertically centered the text inside */
	padding-top: 120px;
	padding-bottom: 120px;
	text-align: center;
	/* applied Google Font */
	font-family: "Open Sans", sans-serif;
}

/* centered the 3 reviewer cards */
.reviewerColumn {
	display: flex;
	flex-direction: column;
	justify-content: center;
	align-items: center;
}

/* removed gutters */
.row1 {
	margin: 0px;
}

.card {
	height: 470px;
}

.cardText {
	padding-left: 5%;
	padding-right: 5%;
	/* applied Google Font */
	font-family: "Roboto Slab", serif;
	font-size: 19px;
}

.cardTitle {
	font-size: 22px;
	font-weight: bold;
	margin-bottom: 20px;
}

/* creates the fade in effect of the heading "Hello. I'm Doctor Smith" */
#fadeInDiv {
	-webkit-animation: fadein 2s; /* supports Safari, Chrome and Opera > 12.1 browsers */
	-moz-animation: fadein 2s; /* supports Firefox < 16 browsers */
	-ms-animation: fadein 2s; /* supports Internet Explorer browsers */
	-o-animation: fadein 2s; /* supports Opera < 12.1 browsers */
	animation: fadein 2s;
}

@keyframes fadein {
	from {
		opacity: 0;
	}
	to {
		opacity: 1;
	}
}

/* Firefox < 16 */
@-moz-keyframes fadein {
	from {
		opacity: 0;
	}
	to {
		opacity: 1;
	}
}

/* Safari, Chrome and Opera > 12.1 */
@-webkit-keyframes fadein {
	from {
		opacity: 0;
	}
	to {
		opacity: 1;
	}
}

/* Internet Explorer */
@-ms-keyframes fadein {
	from {
		opacity: 0;
	}
	to {
		opacity: 1;
	}
}

/* Opera < 12.1 */
@-o-keyframes fadein {
	from {
		opacity: 0;
	}
	to {
		opacity: 1;
	}
}

/* references: 
  https://fonts.google.com/specimen/Padauk
	https://fonts.google.com/specimen/Open+Sans
	https://fonts.google.com/specimen/Roboto+Slab?query=roboto+slab
	https://www.w3schools.com/css/tryit.asp?filename=trycss_align_padding2
	https://stackoverflow.com/questions/11679567/using-css-for-a-fade-in-effect-on-page-load
	
  */
