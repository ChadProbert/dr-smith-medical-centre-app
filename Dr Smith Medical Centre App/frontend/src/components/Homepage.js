import React from "react";
import Button from "@mui/material/Button";
import { Link } from "react-router-dom";
import { Row, Col, Card } from "react-bootstrap";

// components
import ContactUs from "./ContactUs";
import UserNavBar from "./UserNavBar";

// stylesheets
import "./css/Homepage.css";

// images
import reviewer1 from "./images/reviewer images/reviewer 2.jpg";
import reviewer2 from "./images/reviewer images/reviewer 8.jpg";
import reviewer3 from "./images/reviewer images/reviewer 10.jpg";

// basic minimal and reusable homepage
/* features an about us section, a reviews section and multiple links and buttons 
to navigate around the website */
export default function Homepage() {
	return (
		<div className="homepage" data-testid="homepageTest">
			<UserNavBar />
			<div className="background1">
				<Row className="row1">
					<Col md={4}>
						<div>
							<div id="fadeInDiv">
								<h2 className="welcomeHeading">
									Hello. I'm <br /> Dr. Smith
								</h2>
							</div>
						</div>
						<h4 className="welcomeSubHeading">Licensed Medical Doctor</h4>

						<Link to={"/appointments"} className="appointmentsLink">
							<Button variant="contained" className="appointmentsButton">
								View Appointments
							</Button>
						</Link>

						<br />

						<Link to={"/admin-login"} className="adminLoginLink">
							<Button variant="contained" className="adminButton">
								I am an administrator
							</Button>
						</Link>
					</Col>
				</Row>

				<div className="aboutUsBox">
					<h3 className="aboutUsHeading">About Us</h3>
					<p className="aboutUsMessage">
						We are a team of medical health professionals with many years of
						work experience. We pride ourselves on providing the best quality
						healthcare service to our patients.
					</p>
				</div>
			</div>
			<div>
				<div className="reviewsBox">
					<h3 className="reviewsHeading">Reviews</h3>

					<Row className="reviewRow">
						<Col md={4} className="reviewerColumn">
							<Card style={{ width: "22rem" }} className="card">
								<Card.Img variant="top" src={reviewer1} />
								<Card.Body>
									<Card.Title className="cardTitle">Emma Harlow</Card.Title>
									<Card.Text className="cardText">
										"Dr. Smith has been my doctor for 11 years now! She is
										always willing to help. I highly recommend her to anyone who
										is looking for a doctor."
									</Card.Text>
								</Card.Body>
							</Card>
						</Col>

						<Col md={4} className="reviewerColumn">
							<Card style={{ width: "22rem" }} className="card">
								<Card.Img variant="top" src={reviewer2} />
								<Card.Body>
									<Card.Title className="cardTitle">John Doe</Card.Title>
									<Card.Text className="cardText">
										"Dr. Smith is an excellent doctor. She is very passionate as
										well as meticulous when going about doing what she loves!"
									</Card.Text>
								</Card.Body>
							</Card>
						</Col>

						<Col md={4} className="reviewerColumn">
							<Card style={{ width: "22rem" }} className="card">
								<Card.Img variant="top" src={reviewer3} />
								<Card.Body>
									<Card.Title className="cardTitle">Jane Maritz</Card.Title>
									<Card.Text className="cardText">
										"Dr. Smith is a very patient and caring doctor. She is
										always willing to help and is always available to answer any
										questions you may have."
									</Card.Text>
								</Card.Body>
							</Card>
						</Col>
					</Row>
				</div>
			</div>
			<ContactUs />
		</div>
	);
}

/* references: 
	my own work from Level 3 task 9
	https://unsplash.com/
	https://mui.com/
	https://react-bootstrap.github.io/

	*/
