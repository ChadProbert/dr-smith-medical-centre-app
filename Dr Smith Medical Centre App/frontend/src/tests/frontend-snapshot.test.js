import React from "react";
import renderer from "react-test-renderer";
import { MemoryRouter } from "react-router-dom";
import Homepage from "../components/Homepage";

it("Snapshot of Homepage", () => {
	const tree = renderer
		.create(
			<MemoryRouter>
				<Homepage />
			</MemoryRouter>
		)
		.toJSON();

	expect(tree).toMatchSnapshot();
});

/* references: 
  my own work from level 2 task 21
  https://www.youtube.com/watch?v=qOaGoujjc3M&ab_channel=SoftwareTestingHelp
  https://www.npmjs.com/package/react-test-renderer
  
  */
