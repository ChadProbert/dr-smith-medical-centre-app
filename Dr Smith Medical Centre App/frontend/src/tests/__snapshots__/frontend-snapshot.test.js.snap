// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Snapshot of Homepage 1`] = `
<div
  className="homepage"
  data-testid="homepageTest"
>
  <header
    className="MuiPaper-root MuiPaper-elevation MuiPaper-elevation4 MuiAppBar-root MuiAppBar-colorPrimary MuiAppBar-positionStatic appBar css-hip9hq-MuiPaper-root-MuiAppBar-root"
    data-testid="userNavBarTest"
  >
    <div
      className="MuiToolbar-root MuiToolbar-gutters MuiToolbar-regular toolBar css-hyum1k-MuiToolbar-root"
    >
      <a
        href="/"
        onClick={[Function]}
      >
        <img
          alt="logo"
          className="appLogo"
          src="logo.png"
        />
      </a>
      <h6
        className="MuiTypography-root MuiTypography-h6 css-1qp5wg9-MuiTypography-root"
      >
        <a
          className="navBarLink"
          href="/"
          onClick={[Function]}
        >
          Home
        </a>
      </h6>
      <h6
        className="MuiTypography-root MuiTypography-h6 css-1qp5wg9-MuiTypography-root"
      >
        <a
          className="navBarLink"
          href="/appointments"
          onClick={[Function]}
        >
          Appointments
        </a>
      </h6>
      <h6
        className="MuiTypography-root MuiTypography-h6 css-1qp5wg9-MuiTypography-root"
      >
        <a
          className="navBarLink"
          href="/admin-login"
          onClick={[Function]}
        >
          Admin Dashboard
        </a>
      </h6>
    </div>
  </header>
  <div
    className="background1"
  >
    <div
      className="row1 row"
    >
      <div
        className="col-md-4"
      >
        <div>
          <div
            id="fadeInDiv"
          >
            <h2
              className="welcomeHeading"
            >
              Hello. I'm 
              <br />
               Dr. Smith
            </h2>
          </div>
        </div>
        <h4
          className="welcomeSubHeading"
        >
          Licensed Medical Doctor
        </h4>
        <a
          className="appointmentsLink"
          href="/appointments"
          onClick={[Function]}
        >
          <button
            className="MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium MuiButtonBase-root appointmentsButton css-sghohy-MuiButtonBase-root-MuiButton-root"
            disabled={false}
            onBlur={[Function]}
            onContextMenu={[Function]}
            onDragLeave={[Function]}
            onFocus={[Function]}
            onKeyDown={[Function]}
            onKeyUp={[Function]}
            onMouseDown={[Function]}
            onMouseLeave={[Function]}
            onMouseUp={[Function]}
            onTouchEnd={[Function]}
            onTouchMove={[Function]}
            onTouchStart={[Function]}
            tabIndex={0}
            type="button"
          >
            View Appointments
          </button>
        </a>
        <br />
        <a
          className="adminLoginLink"
          href="/admin-login"
          onClick={[Function]}
        >
          <button
            className="MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium MuiButtonBase-root adminButton css-sghohy-MuiButtonBase-root-MuiButton-root"
            disabled={false}
            onBlur={[Function]}
            onContextMenu={[Function]}
            onDragLeave={[Function]}
            onFocus={[Function]}
            onKeyDown={[Function]}
            onKeyUp={[Function]}
            onMouseDown={[Function]}
            onMouseLeave={[Function]}
            onMouseUp={[Function]}
            onTouchEnd={[Function]}
            onTouchMove={[Function]}
            onTouchStart={[Function]}
            tabIndex={0}
            type="button"
          >
            I am an administrator
          </button>
        </a>
      </div>
    </div>
    <div
      className="aboutUsBox"
    >
      <h3
        className="aboutUsHeading"
      >
        About Us
      </h3>
      <p
        className="aboutUsMessage"
      >
        We are a team of medical health professionals with many years of work experience. We pride ourselves on providing the best quality healthcare service to our patients.
      </p>
    </div>
  </div>
  <div>
    <div
      className="reviewsBox"
    >
      <h3
        className="reviewsHeading"
      >
        Reviews
      </h3>
      <div
        className="reviewRow row"
      >
        <div
          className="reviewerColumn col-md-4"
        >
          <div
            className="card card"
            style={
              Object {
                "width": "22rem",
              }
            }
          >
            <img
              className="card-img-top"
              src="reviewer 2.jpg"
            />
            <div
              className="card-body"
            >
              <div
                className="cardTitle card-title h5"
              >
                Emma Harlow
              </div>
              <p
                className="cardText card-text"
              >
                "Dr. Smith has been my doctor for 11 years now! She is always willing to help. I highly recommend her to anyone who is looking for a doctor."
              </p>
            </div>
          </div>
        </div>
        <div
          className="reviewerColumn col-md-4"
        >
          <div
            className="card card"
            style={
              Object {
                "width": "22rem",
              }
            }
          >
            <img
              className="card-img-top"
              src="reviewer 8.jpg"
            />
            <div
              className="card-body"
            >
              <div
                className="cardTitle card-title h5"
              >
                John Doe
              </div>
              <p
                className="cardText card-text"
              >
                "Dr. Smith is an excellent doctor. She is very passionate as well as meticulous when going about doing what she loves!"
              </p>
            </div>
          </div>
        </div>
        <div
          className="reviewerColumn col-md-4"
        >
          <div
            className="card card"
            style={
              Object {
                "width": "22rem",
              }
            }
          >
            <img
              className="card-img-top"
              src="reviewer 10.jpg"
            />
            <div
              className="card-body"
            >
              <div
                className="cardTitle card-title h5"
              >
                Jane Maritz
              </div>
              <p
                className="cardText card-text"
              >
                "Dr. Smith is a very patient and caring doctor. She is always willing to help and is always available to answer any questions you may have."
              </p>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
  <div
    className="contactUsBox"
    data-testid="contactUsTest"
  >
    <h3
      className="contactUsHeading"
    >
      Contact Us To Book An Appointment!
    </h3>
    <p
      className="contactUsText"
    >
      <svg
        aria-hidden={true}
        className="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
        data-testid="CallIcon"
        focusable="false"
        viewBox="0 0 24 24"
      >
        <path
          d="M20.01 15.38c-1.23 0-2.42-.2-3.53-.56-.35-.12-.74-.03-1.01.24l-1.57 1.97c-2.83-1.35-5.48-3.9-6.89-6.83l1.95-1.66c.27-.28.35-.67.24-1.02-.37-1.11-.56-2.3-.56-3.53 0-.54-.45-.99-.99-.99H4.19C3.65 3 3 3.24 3 3.99 3 13.28 10.73 21 20.01 21c.71 0 .99-.63.99-1.18v-3.45c0-.54-.45-.99-.99-.99z"
        />
      </svg>
       012 345 6789
      <br />
      <br />
      <svg
        aria-hidden={true}
        className="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
        data-testid="EmailIcon"
        focusable="false"
        viewBox="0 0 24 24"
      >
        <path
          d="M20 4H4c-1.1 0-1.99.9-1.99 2L2 18c0 1.1.9 2 2 2h16c1.1 0 2-.9 2-2V6c0-1.1-.9-2-2-2zm0 4-8 5-8-5V6l8 5 8-5v2z"
        />
      </svg>
       drsmithmedcentre@gmail.com
      <br />
      <br />
      <svg
        aria-hidden={true}
        className="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
        data-testid="LocationOnIcon"
        focusable="false"
        viewBox="0 0 24 24"
      >
        <path
          d="M12 2C8.13 2 5 5.13 5 9c0 5.25 7 13 7 13s7-7.75 7-13c0-3.87-3.13-7-7-7zm0 9.5c-1.38 0-2.5-1.12-2.5-2.5s1.12-2.5 2.5-2.5 2.5 1.12 2.5 2.5-1.12 2.5-2.5 2.5z"
        />
      </svg>
       123 Sampson Drive, Farrarmere, Benoni, South Africa
    </p>
  </div>
</div>
`;
